compiler-core/src/warning.rs

```diff
@@ -123,11 +123,23 @@ pub enum Warning {
         src: SmolStr,
         warning: crate::type_::Warning,
     },
+    InvalidSource {
+        name: SmolStr,
+    },
 }
 
 impl Warning {
     pub fn to_diagnostic(&self) -> Diagnostic {
         match self {
+            Warning::InvalidSource { name } => Diagnostic {
+                title: "Invalid gleam source filename.".into(),
+                text: "".into(),
```

Comment:
Let's explain what a valid module name is here. The friendlier and more helpful we can make messages from the compiler the better.